### ISSUE TITLE
Loss is always "inf" when running training

### ISSUE THREAD

**Issue by nmd95** [mialbro/PointFusion#1]
![image](https://user-images.githubusercontent.com/46507715/115209225-150d4900-a106-11eb-9aaf-b4b521c1754d.png)

 
Hi, I tried to run train.py after cloning the repo + downloading the dataset. As can be seen in the attached screen-shot, the loss remains "inf" even after approx 200 epochs into training.

Is there something to do about it ? what is going on here ?

Any help would be much appreciated !

Cheers,

**Comment by nmd95** [comment]
This seems to only be resolved by reducing the batch-size to 1.

**Comment by mialbro** [comment]
Ok. I will take a look and get back to you on this problem.


________________________________
From: nmd95 ***@***.***>
Sent: Friday, April 23, 2021 1:05:09 PM
To: mialbro/PointFusion ***@***.***>
Cc: Robinson, Mark D. ***@***.***>; Comment ***@***.***>
Subject: [EXT] Re: [mialbro/PointFusion] Loss is always "inf" when running training (#1)


This seems to only be resolved by reducing the batch-size to 1.

**Comment by mialbro** [comment]
Caused by log(0) Fixed by adding epsilon to log in unsupervised loss function `loss = ((loss_offset * pred_scores) - (weight * torch.log(pred_scores + eps)))`
